### PR TITLE
Bug fix: Logic for entering a service ID value larger than MAX_SERVICE 

### PR DESCRIPTION
**Pull request by dennisafa**
Fixes a bug related to entering a service ID larger than the defined macro MAX_SERVICE.  

**Summary:**
Upon entering a service ID greater than 31 as a command line argument in the initialization of a NF, an open manager would crash. This references this opened [issue. ](https://github.com/sdnfv/openNetVM/issues/64)
**Test Plan:**
Tested on speed_tester. Plan to test on other NF's for thoroughness.

### CODE REVIEW

**Comment by dennisafa** [comment]
I'm working on adding a NF_MAX_SERVICES definition that'll represent that the service id value was exceeded instead of simply setting the status to NF_NO_IDS

File: onvm/onvm_nflib/onvm_nflib.c [hunk 1]
```diff
@@ -315,6 +315,9 @@ onvm_nflib_start_nf(struct onvm_nf_info *nf_info) {
         if (nf_info->status == NF_ID_CONFLICT) {
                 rte_mempool_put(nf_info_mp, nf_info);
                 rte_exit(NF_ID_CONFLICT, "Selected ID already in use. Exiting...\n");
+        } else if (nf_info->status == NF_SERVICE_MAX) {
+                rte_mempool_put(nf_info_mp, nf_info);
+                rte_exit(NF_SERVICE_MAX, "Service ID has exceeded the maximum amount\n");
```
Comment (twood02) [inline onvm/onvm_nflib/onvm_nflib.c]:
```suggestion
                rte_exit(NF_SERVICE_MAX, "Service ID must be less than %d\n", MAX_SERVICES);
```

**Comment by twood02** [comment]
- Sanity checks, assigned to @kevindweb 
    - run make in `/onvm` and `/examples` directories

- Code style, assigned to @kevindweb
    - Run linter
    - Check pretty much everything style related

- Code design, assigned to @kevindweb
    - Check for core reusability
    - Code is clean, functions are concise
    - Check for memory leaks
    - Edge cases properly handled

- Performance, assigned to @kevindweb
    - Run Speed tester NF, report performance.

- Documentation, assigned to @kevindweb
    - Check if the new changes are well documented, in both code and READMEs

**Comment by nks5295** [comment]
i'll review this through the weekend. please comment when the assigned items above are completed.

**Comment by kevindweb** [comment]
Good work @dennisafa , here's my analysis

- Everything built and ran correctly
`~/openNetVM/examples/speed_tester$ ./go.sh 5 34 34 -c 1600`
Provided this error correctly and exited the process
`EAL: Error - exiting with code: 7
  Cause: Service ID must be less than 32`
This correct case worked with similar performance to previous trials without the updated case
`~/openNetVM/examples/speed_tester$ ./go.sh 5 34 34 -c 1600`
Average Tx packets per second on my instance: ~44870000+

- Error checking was placed where all other nf_info->service_id errors were checked for (reusability/scalability) correctly

- No errors showing from gwclint.py from the edited lines in the any of the 3 files committed
- Comment was made where applicable, and a good exit error message

**Comment by twood02** [comment]
As part of this PR we should sanity check other cases -- what happens if you try to use service ID `-10`? `0`?  Is `32` allowed or is the max `31`?

Are those already handled or do we need to add to this?

**Comment by dennisafa** [comment]
@twood02 will check those out now and commit as necessary.

**Comment by dennisafa** [comment]
@twood02 31 is fine, but 32 breaks it. that's in place. a nonzero/negative service id check already exists as well.

File: onvm/onvm_mgr/onvm_nf.c [hunk 2]
```diff
@@ -188,11 +188,20 @@ onvm_nf_start(struct onvm_nf_info *nf_info) {
                 return 1;
         }
 
+        if (nf_info->service_id >= MAX_SERVICES) {
+                // Service ID must be less than MAX_SERVICES and greater than 0
+                nf_info->status = NF_SERVICE_MAX;
+                return 1;
+        }
+
+
```
Comment (koolzz) [inline onvm/onvm_mgr/onvm_nf.c]: As we're fixing bugs I think we can address another 🐛.
If we spawn more than MAX_NFS_PER_SERVICE NFs with service id X we're going to crash the manager when accessing the `nf_per_service_count` array later on.

Basically add another check that exits if  `nf_per_service_count[nf_info->service_id] >= MAX_NFS_PER_SERVICE`, Add the new error type as well that was a good idea.

**Comment by ratnadeepb** [comment]
@dennisafa NF_Dev.md document says "Currently, our platform supports at most 16 NF instances running at once. This limit is defined in onvm_common.h.". However, this limit has already been updated to 128 on the onvm_common.h file.

**Comment by koolzz** [comment]
> @dennisafa NF_Dev.md document says "Currently, our platform supports at most 16 NF instances running at once. This limit is defined in onvm_common.h.". However, this limit has already been updated to 128 on the onvm_common.h file.

@dennisafa Following up on that, lets clear up how many NFs can we run in total, how many service ids we can have and how many instances per service id can be spawned. Also add a line that those parameters could be changed, and this is just what we use for our experiments.

File: onvm/onvm_mgr/onvm_nf.c (hunk 2 above)
Comment (dennisafa) [reply onvm/onvm_mgr/onvm_nf.c]: Sounds good, I added that check and created a new error type (8) for this scenario.

**Pull request merged**